lib/thrift/binary/framed/protocol_handler.ex

```diff
@@ -5,20 +5,26 @@ defmodule Thrift.Binary.Framed.ProtocolHandler do
 
   @default_timeout 20_000
 
+  @typedoc "A module that implements the :ranch_transport behaviour"
+  @type transport :: :ranch_tcp | :ranch_ssl
+
+  @typedoc "Transport-specific options"
+  @type transport_opts :: :ranch_tcp.opts | :ranch_ssl.opts
+
   alias Thrift.{
     Protocol,
     TApplicationException
   }
   require Logger
 
-  @spec start_link(reference, pid, module, {module, module, Keyword.t}) :: GenServer.on_start
-  def start_link(ref, socket, transport, {server_module, handler_module, tcp_opts}) do
-    pid = spawn_link(__MODULE__, :init, [ref, socket, transport, server_module, handler_module, tcp_opts])
+  @spec start_link(reference, port, transport, {module, module, transport_opts}) :: GenServer.on_start
+  def start_link(ref, socket, transport, {server_module, handler_module, transport_opts}) do
+    pid = spawn_link(__MODULE__, :init, [ref, socket, transport, server_module, handler_module, transport_opts])
     {:ok, pid}
   end
 
-  @spec init(reference, pid, module, module, module, Keyword.t) :: :ok | no_return
-  def init(ref, socket, transport, server_module, handler_module, tcp_opts) do
+  @spec init(reference, port, :ranch_tcp, module, module, :ranch_tcp.opts) :: :ok | no_return
```

Comment:
Ah I didn't realize that.

> I guess, given that, I should drop :ranch_ssl from the transport type for now?

Yeah that makes sense to me.  Maybe an issue so someone remembers to go back and support it?